### PR TITLE
Changed index in Target

### PR DESCRIPTION
**Pull request by andresg4**
#### Description:

* Allow targets have the same title for different users.
---
#### Tasks Done

- [X] Removed previous index in Target.
- [X] Created new index including title and user_id.
- [X] Changed validation in Target model.
- [X] Modified tests.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -29,6 +29,27 @@
     t.index ["resource_type", "resource_id"], name: "index_active_admin_comments_on_resource_type_and_resource_id"
   end
 
+  create_table "active_storage_attachments", force: :cascade do |t|
```
Comment (Alxzu) [inline db/schema.rb]: This code shouldn't be here, since this is for active storage

File: db/schema.rb [hunk 2]
```diff
@@ -46,6 +67,18 @@
     t.index ["reset_password_token"], name: "index_admin_users_on_reset_password_token", unique: true
   end
 
+  create_table "administrators", force: :cascade do |t|
```
Comment (Alxzu) [inline db/schema.rb]: This code shouldn't be here in this PR

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -58,14 +65,54 @@
       expect(json['match_targets']).to be_empty
     end
 
+    context 'same title but from other user' do
+      before do
+        params[:target][:title] = target_same_title.title
+        # post api_v1_targets_path, params: params,
```
Comment (Alxzu) [inline spec/requests/api/v1/targets/create_spec.rb]: Remove commented code

File: spec/requests/api/v1/targets/create_spec.rb [hunk 4]
```diff
@@ -6,16 +6,23 @@
...
+  let(:topic)              { create(:topic) }
+  let(:topic_other_target) { create(:topic) }
 
   let(:target) do
     create(:target, user: user_match, latitude: params[:target][:latitude] + 0.0001,
                     longitude: params[:target][:longitude] + 0.0001,
                     topic_id: params[:target][:topic_id])
   end
 
+  let(:target_same_title) do
```
Comment (jdrosales17) [inline spec/requests/api/v1/targets/create_spec.rb]: I would move this `let` inside the context where it is used (`context 'same title but from other user'`)

**Review by jdrosales17 — APPROVED** [review]

File: spec/requests/api/v1/targets/create_spec.rb [hunk 5]
```diff
@@ -58,14 +65,51 @@
...
+        expect(new_target.radius).to eq(params[:target][:radius].to_f)
+        expect(new_target.latitude.to_f).to eq(params[:target][:latitude].round(6))
+        expect(new_target.longitude.to_f).to eq(params[:target][:longitude].round(6))
+        expect(new_target.topic_id).to eq(params[:target][:topic_id])
+      end
+
+      it 'returns target information' do
+        subject
+        new_target = user.targets.find_by_title(params[:target][:title])
+        expect(json['id']).to eq(new_target.id)
```
Comment (jdrosales17) [inline spec/requests/api/v1/targets/create_spec.rb:94]: You can use `rspec/json_expectations` gem which would compare the whole JSON in one `expect`, just a suggestion :)

[Here](https://relishapp.com/waterlink/rspec-json-expectations/docs/json-expectations/include-json-matcher) you can find some examples

Comment (andresg4) [reply spec/requests/api/v1/targets/create_spec.rb:94]: Great suggestion. I will do this in other PR.

File: spec/models/target_spec.rb [hunk 6]
```diff
@@ -6,7 +6,9 @@
   describe 'attributes' do
     it { is_expected.to validate_presence_of(:title) }
     it { is_expected.to validate_length_of(:title) }
-    it { is_expected.to validate_uniqueness_of(:title).ignoring_case_sensitivity }
+    it {
```
Comment (Alxzu) [inline spec/models/target_spec.rb]: If you are doing multiple rows use `do` `end` instead of brackets

**Review by Alxzu — APPROVED** [review]
Small comment

**Pull request merged**